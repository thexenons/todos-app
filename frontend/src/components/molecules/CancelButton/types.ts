import { ButtonProps } from "../../atoms/Button/types";

export interface CancelButtonProps {
	disabled?: ButtonProps["disabled"];
}
