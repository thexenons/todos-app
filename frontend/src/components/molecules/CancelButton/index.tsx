import { FC } from "react";

import Button from "../../atoms/Button";
import { ButtonColorVariants, ButtonVariants } from "../../atoms/Button/types";
import { CancelButtonProps } from "./types";

const CancelButton: FC<CancelButtonProps> = ({ disabled = false }) => (
	<Button
		variant={ButtonVariants.contained}
		colorVariant={ButtonColorVariants.neutral}
		disabled={disabled}
	>
		Cancel
	</Button>
);

export default CancelButton;
