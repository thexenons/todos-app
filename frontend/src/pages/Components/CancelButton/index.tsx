import { FC } from "react";

import CancelButton from "../../../components/molecules/CancelButton";
import * as SS from "../styled";

const CancelButtonComponenstPage: FC = () => (
	<SS.ComponentsListWrapper>
		<h1>CancelButton</h1>
		<SS.ComponentsRowWrapper>
			<CancelButton />
			<CancelButton disabled />
		</SS.ComponentsRowWrapper>
	</SS.ComponentsListWrapper>
);

export default CancelButtonComponenstPage;
