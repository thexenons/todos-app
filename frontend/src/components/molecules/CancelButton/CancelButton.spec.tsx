import { render } from "@testing-library/react";

import BaseTestComponent from "../../../tests/BaseTestComponent";
import CancelButton from ".";

describe("CancelButton", () => {
	it("Render", () => {
		const { container } = render(
			<BaseTestComponent>
				<CancelButton />
			</BaseTestComponent>
		);
		expect(container).toMatchSnapshot();
	});
});
