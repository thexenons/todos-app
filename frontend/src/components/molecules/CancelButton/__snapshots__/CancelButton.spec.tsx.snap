// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CancelButton > Render 1`] = `
<div>
  <button
    class="styled__ButtonWrapper-sc-f8cd241f-0 bSHSQa"
  >
    Cancel
  </button>
</div>
`;
